### PR TITLE
cluster deploy extension: Add localization support and fix " to ' strings

### PR DESCRIPTION
**Pull request by kevcunnane**
As part of learning the codebase I spotted a few issues:
- We should localize all strings. Converted to expected nls.localize format
- We get compile warnings due to tslint if we use " instead of '. Fixed
- Converted 1 usage of \n to use os-specific EOL.

### CODE REVIEW

File: extensions/big-data-cluster/src/installer/installer.ts [hunk 1]
```diff
@@ -29,7 +32,7 @@ export async function installKubectl(shell: Shell): Promise<Errorable<null>> {
     const downloadFile = path.join(installFolder, binFile);
     const downloadResult = await download.to(kubectlUrl, downloadFile);
     if (failed(downloadResult)) {
-        return { succeeded: false, error: [`Failed to download kubectl: ${downloadResult.error[0]}`] };
+        return { succeeded: false, error: [localize('downloadKubectlFailed', 'Failed to download kubectl: ${0}', downloadResult.error[0])] };
```
Comment (alanrenmsft) [inline extensions/big-data-cluster/src/installer/installer.ts]: I think you have to remove the dollor sign here. previous it was using template string but not it is not required

File: extensions/big-data-cluster/src/installer/installer.ts [hunk 2]
```diff
@@ -43,7 +46,7 @@ export async function installKubectl(shell: Shell): Promise<Errorable<null>> {
 async function getStableKubectlVersion(): Promise<Errorable<string>> {
     const downloadResult = await download.toTempFile('https://storage.googleapis.com/kubernetes-release/release/stable.txt');
     if (failed(downloadResult)) {
-        return { succeeded: false, error: [`Failed to establish kubectl stable version: ${downloadResult.error[0]}`] };
+        return { succeeded: false, error: [localize('kubectlVersionCheckFailed', 'Failed to establish kubectl stable version: ${0}', downloadResult.error[0])] };
```
Comment (alanrenmsft) [inline extensions/big-data-cluster/src/installer/installer.ts]: same here, remove $

File: extensions/big-data-cluster/src/kubectl/kubectl.ts [hunk 3]
```diff
@@ -72,17 +75,17 @@ async function checkForKubectlInternal(context: Context, errorMessageMode: Check
     const bin = getToolPath(context.host, context.shell, binName);
 
     const contextMessage = getCheckKubectlContextMessage(errorMessageMode);
-    const inferFailedMessage = `Could not find "${binName}" binary.${contextMessage}`;
-    const configuredFileMissingMessage = `${bin} is not installed. ${contextMessage}`;
+    const inferFailedMessage = localize('binaryNotFound', 'Could not find {0} binary. {1}', binName, contextMessage);
+    const configuredFileMissingMessage = localize('binaryNotInstalled', '${0} is not installed. ${1}', bin, contextMessage);
```
Comment (alanrenmsft) [inline extensions/big-data-cluster/src/kubectl/kubectl.ts]: same here, remove $

Comment (kevcunnane) [reply extensions/big-data-cluster/src/kubectl/kubectl.ts]: Whoops - will fix all of the many issues with this you spotted. Thanks!

**Pull request merged**